src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1724 @@
+			Lambna = 2260; //(kJ/kg) latent heat of vaporization ePlus should carry this an available variable
+			count_EnvironmentConditionsMetOnce = 0;
+			count_EnvironmentConditionsNotMet = 0;
+			count_SAHR_OC_MetOnce = 0;
+			count_SAT_OC_MetOnce = 0;
+			count_DidWeMeetLoad = 0;
+			count_DidWeNotMeetLoad = 0;
+			// vector below used store the modes in each timestep that don't meet humidity or temperature limits, used in warnings
+			vector<int> temp(25);
+			SAT_OC_MetinMode_v= temp;
+			SAHR_OC_MetinMode_v= temp;
+			
+			int MAXIMUM_OPERATIONAL_SETTINGS = 5;
+			ModeCounter = 0;
+			//these are the three setting types currently implimented, pOptimal is the setting that meets conditioning load. 
+			pStandBy = new CSetting;
+			pOptimal= new CSetting;
+			
+
+			for (int i = 0; i< MAXIMUM_OPERATIONAL_SETTINGS; i++)
+			{
+				CSetting * pointer= new CSetting();
+				pointer->ElectricalPower = 0;
+				CurrentOperatingSettings.push_back(pointer);
+			}
+			InitializeModelParams();
+		}
+		
+		void Model::ResetOutputs()
+		{
+			UnitTotalCoolingRate = 0;
+			UnitTotalCoolingEnergy = 0;
+			UnitSensibleCoolingRate = 0;
+			UnitSensibleCoolingEnergy = 0;
+			UnitLatentCoolingRate = 0;
+			UnitLatentCoolingEnergy = 0;
+			SystemTotalCoolingRate = 0;
+			SystemTotalCoolingEnergy = 0;
+			SystemSensibleCoolingRate = 0;
+			SystemSensibleCoolingEnergy = 0;
+			SystemLatentCoolingRate = 0;
+			SystemLatentCoolingEnergy = 0;
+			UnitTotalHeatingRate = 0;
+			UnitTotalHeatingEnergy = 0;
+			UnitSensibleHeatingRate = 0;
+			UnitSensibleHeatingEnergy = 0;
+			UnitLatentHeatingRate = 0;
+			UnitLatentHeatingEnergy = 0;
+			SystemTotalHeatingRate = 0;
+			SystemTotalHeatingEnergy = 0;
+			SystemSensibleHeatingRate = 0;
+			SystemSensibleHeatingEnergy = 0;
+			SystemLatentHeatingRate = 0;
+			SystemLatentHeatingEnergy = 0;
+			RequestedLoadToCoolingSetpoint = 0;
+			RequestedHumdificationMass = 0;
+			RequestedHumdificationLoad = 0;
+			RequestedHumdificationEnergy = 0;
+			RequestedDeHumdificationMass = 0;
+			RequestedDeHumdificationLoad = 0;
+			RequestedDeHumdificationEnergy = 0;
+			PrimaryMode = 0;
+			PrimaryModeRuntimeFraction = 0;
+
+		}
+
+		void Model::InitializeModelParams()
+		{
+			ResetOutputs();
+			optimal_EnvCondMet = false;
+			Tsa = 0;
+			//reset the power use to a high value, this is replaced during the calculation keeping the "best" setting. 
+			pOptimal->ElectricalPower = IMPLAUSIBLE_POWER;
+			RunningPeakCapacity_EnvCondMet = false;
+			Settings.clear();
+		}
+
+		Model::~Model()                 
+		{
+			list<CModeSolutionSpace*>::iterator iter;
+			for (iter = SolutionSpaces.begin(); iter != SolutionSpaces.end(); ++iter) {
+				CModeSolutionSpace* p = (*iter);
+				delete p;
+			}
+			
+			list<CSetting*>::iterator iterSettings;
+			for (iterSettings = Settings.begin(); iterSettings != Settings.end(); ++iterSettings) {
+				CSetting* ps = (*iterSettings);
+				delete ps;
+			}
+			CurrentOperatingSettings.clear();
```

Comment:
Yes all the settings are in another container so I don't want to delete them here.